pydra/engine/tests/test_task.py

```diff
@@ -1028,21 +1047,92 @@ def test_audit_shellcommandtask(tmpdir):
     shelly()
     message_path = tmpdir / shelly.checksum / "messages"
     # go through each jsonld file in message_path and check if the label field exists
-    label_content = []
+
     command_content = []
 
     for file in glob(str(message_path) + "/*.jsonld"):
         with open(file, "r") as f:
             data = json.load(f)
-            print(data)
-            if "Label" in data:
-                label_content.append(True)
+
+            if "@type" in data:
+                if "AssociatedWith" in data:
+                    assert "shelly" in data["Label"]
+
+            if "@type" in data:
+                if data["@type"] == "input":
+                    assert data["Label"] == None
+
             if "Command" in data:
                 command_content.append(True)
                 assert "ls -l" == data["Command"]
 
-    print(command_content)
-    assert any(label_content)
+    assert any(command_content)
+
+
+def test_audit_shellcommandtask_file(tmpdir):
+    import shutil
+
+    # create test.txt file with "This is a test" in it in the tmpdir
+    with open(tmpdir / "test.txt", "w") as f:
+        f.write("This is a test.")
+    # make a copy of the test.txt file in the tmpdir and name it test2.txt
+    shutil.copy(tmpdir / "test.txt", tmpdir / "test2.txt")
+
+    cmd = "cat"
+    file_in = tmpdir / "test.txt"
+    file_in_2 = tmpdir / "test2.txt"
+    test_file_hash = hash_file(file_in)
+    test_file_hash_2 = hash_file(file_in_2)
+    my_input_spec = SpecInfo(
+        name="Input",
+        fields=[
+            (
+                "in_file",
+                attr.ib(
+                    type=File,
+                    metadata={
+                        "position": 1,
+                        "argstr": "",
+                        "help_string": "text",
+                        "mandatory": True,
+                    },
+                ),
+            ),
+            (
+                "in_file_2",
+                attr.ib(
+                    type=File,
+                    metadata={
+                        "position": 2,
+                        "argstr": "",
+                        "help_string": "text",
+                        "mandatory": True,
+                    },
+                ),
+            ),
+        ],
+        bases=(ShellSpec,),
+    )
+    shelly = ShellCommandTask(
+        name="shelly",
+        in_file=file_in,
+        in_file_2=file_in_2,
+        input_spec=my_input_spec,
+        executable=cmd,
+        audit_flags=AuditFlag.PROV,
+        messengers=PrintMessenger(),
+    )
+    shelly.cache_dir = tmpdir
+    shelly()
+    message_path = tmpdir / shelly.checksum / "messages"
+    for file in glob.glob(str(message_path) + "/*.jsonld"):
+        with open(file, "r") as f:
+            data = json.load(f)
+            print(file_in)
+            if "AtLocation" in data:
+                assert data["AtLocation"] == [file_in, file_in_2]
```

Comment:
I'm not sure why you expect list here. But the test pass because it never gets here!

Reply:
WIP